### PR TITLE
Add support for Volume creation

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**

- Check external filesystems exists on AWS on `up`
- Search for a previous filesystem associated with project's volume
- Create new filesystems for volume if needed
- Create TargetGroups within project's SecurityGroup to allow NFS mounts from services

Major change here is that we don't assume anymore EFS filesystem has TargetGroups we need to attach to by backing the adequate ingress rules. We create TargetGroups inside project's SecurityGroup (arbitrary traffic is already allowed inside project's SecurityGroup). Those can be deleted by `compose down` without the filesystem been impacted.

**usage**
```yaml
services:
  jenkins:
    image: jenkins/jenkins:lts
    volumes:
      - config:/var/jenkins_home
volumes:
  config: {}
```

First `compose up` will create a filesystem on EFS and attach as volume to the container
volume will be re-used by next `compose up` based on tags (project name + volume name)

#close https://github.com/docker/compose-cli/issues/737
#close https://github.com/docker/compose-cli/issues/738

/test-ecs

image

### CODE REVIEW

File: ecs/awsResources.go [hunk 1]
```diff
@@ -139,30 +141,52 @@ func (b *ecsAPIService) parseLoadBalancerExtension(ctx context.Context, project
...
+		if !net.External.External {
+			continue
+		}
+		exists, err := b.SDK.SecurityGroupExists(ctx, net.Name)
 		if err != nil {
 			return nil, err
 		}
 		if !exists {
-			return nil, fmt.Errorf("security group %s doesn't exist", sg)
+			return nil, fmt.Errorf("security group %s doesn't exist", net.Name)
```
Comment (chris-crone) [inline ecs/awsResources.go:164]:
```suggestion
			return nil, errors.Wrapf(errdefs. ErrNotFound, "security group %q", net.Name)
```

File: ecs/awsResources.go [hunk 2]
```diff
@@ -139,30 +141,52 @@ func (b *ecsAPIService) parseLoadBalancerExtension(ctx context.Context, project
...
+	for name, vol := range project.Volumes {
+		if !vol.External.External {
+			continue
+		}
+		exists, err := b.SDK.FileSystemExists(ctx, vol.Name)
+		if err != nil {
+			return nil, err
+		}
+		if !exists {
+			return nil, fmt.Errorf("EFS file system %s doesn't exist", vol.Name)
```
Comment (chris-crone) [inline ecs/awsResources.go:183]:
```suggestion
			return nil, errors.Wrapf(errdefs.ErrNotFound, "EFS file system %q", vol.Name)
```

File: ecs/awsResources.go [hunk 3]
```diff
@@ -20,13 +20,15 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/aws/aws-sdk-go/service/elbv2"
-	"github.com/awslabs/goformation/v4/cloudformation/ec2"
-	"github.com/awslabs/goformation/v4/cloudformation/elasticloadbalancingv2"
+	"github.com/docker/compose-cli/errdefs"
```
Comment (chris-crone) [inline ecs/awsResources.go:23]: goimports

File: ecs/cloudformation.go [hunk 4]
```diff
@@ -23,6 +23,10 @@ import (
 	"regexp"
 	"strings"
 
+	"github.com/sirupsen/logrus"
+
+	"github.com/docker/compose-cli/api/compose"
```
Comment (chris-crone) [inline ecs/cloudformation.go:28]: goimports

**Review by aiordache — APPROVED** [review]
Deployed a compose file with a named volume and it worked fine. Didn't spot anything wrong with the code.

File: ecs/marshall.go [hunk 5]
```diff
@@ -39,7 +39,8 @@ func marshall(template *cloudformation.Template) ([]byte, error) {
 		if resources, ok := input["Resources"]; ok {
 			for _, uresource := range resources.(map[string]interface{}) {
 				if resource, ok := uresource.(map[string]interface{}); ok {
-					if resource["Type"] == "AWS::ECS::TaskDefinition" {
+					switch resource["Type"] {
+					case "AWS::ECS::TaskDefinition":
```
Comment (gtardif) [inline ecs/marshall.go:43]: changing this for a switch, because you foresee we'll have other cases ? (just curious)

File: ecs/sdk.go [hunk 6]
```diff
@@ -71,6 +71,11 @@ func newSDK(sess *session.Session) sdk {
 	sess.Handlers.Build.PushBack(func(r *request.Request) {
 		request.AddToUserAgent(r, "Docker CLI")
 	})
+	/*
+		sess.Config.LogLevel = aws.LogLevel(
+			aws.LogDebugWithRequestErrors | aws.LogDebugWithRequestRetries | aws.LogDebugWithHTTPBody,
+		)
+	*/
```
Comment (gtardif) [inline ecs/sdk.go:78]: we can remove this comment I guess

**Review by gtardif — APPROVED** [review]
1 nit 1 question, but LGTM

File: ecs/marshall.go (hunk 5 above)
Comment (ndeloof) [reply ecs/marshall.go:43]: yes indeed. goformation can't handle negative boolean (default=true)
For testing purpose I added a few other cases. Using a switch then has less impacts

**Pull request merged**